android/src/main/kotlin/com/apptentive/apptentive_flutter/ApptentiveFlutterPlugin.kt

```diff
@@ -18,27 +14,30 @@ import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 
+
+@OptIn(InternalUseOnly::class)
 class ApptentiveFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
 
   // The MethodChannel that will communicate between Flutter and native Android
   // This local reference serves to register the plugin with the Flutter Engine
   // and unregister it when the Flutter Engine is detached from the Activity
   private lateinit var channel : MethodChannel
-
-  // Current Application object
-  private var application: Application? = null
-
-  // Current Activity object
+  private lateinit var application: Application
   private var activity: Activity? = null
-
-  // Result error code
   private val ERROR_CODE: String = "Apptentive Error"
+  private var isApptentiveRegistered: Boolean = false
 
 
+  private val activityInfo = object : ApptentiveActivityInfo {
+    override fun getApptentiveActivityInfo(): Activity? {
+      if (activity == null) {
+        Log.w(LogTag("Flutter"), "Activity should not be null")
```

Comment:
Should this be an error? Would anything work if the activity is null?

Reply:
We don't engage if the activity is `null`.

Reply:
swap this to error.